src/Chainweb/Pact/PactService.hs

```diff
@@ -295,8 +287,23 @@ serviceRequests memPoolAccess reqQ = do
                     execLookupPactTxs restorePoint txHashes
                 go
             PreInsertCheckMsg (PreInsertCheckReq txs resultVar) -> do
+
+                -- FIXME what can we use as parent hash here?
+                -- PublicData has "", but that's not a valid BlockHash.
+                --
+                -- For now grab the last header from the block hash, but that
+                -- can be @Nothing@.
+                -- Or should we use the consensus state?
+                --
+                cp <- view $ psCheckpointEnv . cpeCheckpointer
+                latestHash <- liftIO (_cpGetLatestBlock cp) >>= \case
+                    Nothing -> fmap _blockHash $ genesisBlockHeader
+                        <$> view chainwebVersion
+                        <*> view chainId
+                    Just (_, x) -> return x
```

Comment:
@emilypi, @gregorycollins, is this the right thing to do? Please comment.

Reply:
This is costly, but I'll have a fix out immediately after this removing it by tracking the previous block's parent hash in state. For the meantime, this is fine.